### PR TITLE
Add argument to allow limiting docker container to s specific amount of memory

### PR DESCRIPTION
**Pull request by sivel**
##### SUMMARY
Add argument to allow limiting docker container to s specific amount of memory
##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
test/runner

##### ANSIBLE VERSION

```
devel
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: test/runner/test.py [hunk 1]
```diff
@@ -605,6 +605,9 @@ def add_extra_docker_options(parser, integration=True):
                         action='store_true',
                         help='run docker container in privileged mode')
 
+    docker.add_argument('--docker-memory',
+                        help='Memory limit for docker in bytes', type=int)
```
Comment (mattclay) [inline test/runner/test.py]: Use lowercase `memory` for consistency with other options.

File: test/runner/lib/delegation.py [hunk 2]
```diff
@@ -168,6 +169,8 @@ def delegate_docker(args, exclude, require):
         '--docker-privileged': 0,
         '--docker-util': 1,
     }
+    if memory_limit:
+        options['--docker-memory'] = memory_limit
```
Comment (mattclay) [inline test/runner/lib/delegation.py]: These two lines are not needed, as there is no reason to filter out the option when delegating.

File: test/runner/lib/delegation.py [hunk 3]
```diff
@@ -154,6 +154,7 @@ def delegate_docker(args, exclude, require):
     util_image = args.docker_util
     test_image = args.docker
     privileged = args.docker_privileged
+    memory_limit = args.docker_memory
```
Comment (mattclay) [inline test/runner/lib/delegation.py]: No need to copy this to a local variable.

Yes, the existing code does that for a few args, but it's unnecessary. No need to fix those though.

**Review by mattclay — CHANGES_REQUESTED** [review]

**Comment by sivel** [comment]
@mattclay review comments addressed

**Review by mattclay — APPROVED** [review]

**Pull request merged**